docs-src/prod-readiness-context/how-to-set-up-codec-server.md

```diff
@@ -1,34 +1,230 @@
 ---
 id: how-to-set-up-codec-server
-title: Setting Codec Server endpoints
-sidebar_label: Setting Codec Server endpoints
-description: Run a Codec Server with your Payload Codec and then configure tctl and the Web UI to use the server.
+title: Codec Server Setup
+sidebar_label: Codec Server Setup
+description: Run a Codec Server with your Payload Codec and then configure CLI and the Web UI to use the server endpoints.
 tags:
   - security
   - how-to
 ---
 
-To use a [Codec Server](/concepts/what-is-a-codec-server), first run it with your Payload Codec and then configure [tctl](/tctl-v1) and the [Web UI](/web-ui) to use it.
+Use a Codec Server to decrypt your encoded [payloads](/concepts/what-is-a-payload) and integrate it with the Temporal Web UI and CLI commands when debugging your Workflows.
 
-#### tctl
+A Codec Server is an HTTP or HTTPS Server that you create and host.
+It must be configured to use a [Payload Codec](/concepts/what-is-a-payload-codec) with the required decode logic and encryption keys.
 
-After the Codec Server is started, provide the exposed endpoint to tctl using the `--codec_endpoint` global option.
+The Codec Server is independent of the Temporal Server and decodes your encrypted payloads through endpoints.
+When you set the codec endpoint in the Temporal Web UI, the Web UI uses the remote endpoint to send encoded data to the Codec Server and receive decoded payloads from the Codec Server. See [API contract requirements](#api-contract-specifications).
+Decoded payloads are then displayed in the Workflow Execution Event History on the Web UI.
+Note that the decoded payloads are visible only to you on the Web UI; All data on the Temporal Server remains encrypted.
```

Comment:
```suggestion
Note that the decoded payloads are visible only to you on the Web UI; all data on the Temporal Server remains encrypted.
```

Reply:
Changed to "Note that when you use a Codec Server, the decoded payloads are visible only to you on the Web UI; payloads on the Temporal Server (whether on Temporal Cloud or a self-hosted Temporal Cluster) remain encrypted."